config/config_test.go

```diff
@@ -35,3 +42,56 @@ func (s *configSuite) TestPrefixed(c *C) {
 	c.Assert(s.tlc.policy("quux"), Equals, s.tlc.prefixed(rootPolicy, "quux"))
 	c.Assert(s.tlc.volume("foo", "bar", "quux"), Equals, s.tlc.prefixed(rootVolume, "foo", "bar", "quux"))
 }
+
+func (s *configSuite) TestDumpTarball(c *C) {
+	key := "/volplugin/foo"
+	value := "baz"
+
+	exec.Command("/bin/sh", "-c", "etcdctl set "+key+" "+value).Run()
+	defer exec.Command("/bin/sh", "-c", "etcdctl rm "+key).Run()
+
+	tarballPath, err := s.tlc.DumpTarball()
+	c.Assert(err, IsNil)
+	defer os.Remove(tarballPath)
+
+	// check that the tarball was created in the expected spot
+	_, err = os.Stat(tarballPath)
+	c.Assert(os.IsNotExist(err), Equals, false)
+
+	parts := strings.Split(tarballPath, "/")
+	tarballFilename := parts[len(parts)-1]
+	dirName := tarballFilename[:strings.LastIndex(tarballFilename, ".tar.gz")]
+
+	testFilename := dirName + key
+	_ = testFilename
+
+	data, err := ioutil.ReadFile(tarballPath)
+	c.Assert(err, IsNil)
+
+	// check that our test key is present and has the expected value
+	reader := bytes.NewReader(data)
+	gzReader, err := gzip.NewReader(reader)
+	c.Assert(err, IsNil)
+	tarReader := tar.NewReader(gzReader)
+
+	found := false
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		c.Assert(err, IsNil)
+
+		if header.Name == testFilename {
+			var b bytes.Buffer
+			_, err = io.Copy(&b, tarReader)
+			c.Assert(err, IsNil)
```

Comment:
According to [the docs][1]:

```
A successful Copy returns err == nil, not err == EOF. Because Copy is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported.
```

[1]: https://golang.org/pkg/io/#Copy